src/cloud-api/src/client/cloud_provider.rs

```diff
@@ -0,0 +1,162 @@
+        match self.id.as_str() {
+            "aws/us-east-1" => Ok(CloudProviderRegion::AwsUsEast1),
+            "aws/eu-west-1" => Ok(CloudProviderRegion::AwsEuWest1),
+            _ => Err(Error::CloudProviderRegionParseError),
+        }
+    }
+
+    /// Returns the region controller endpoint subdomain
+    pub fn rc_subdomain(&self) -> String {
+        let host = self.api_url.host().unwrap().to_string();
+        let index = host.find("cloud.materialize.com").unwrap();
+        let subdomain: String = host[..index - 1].to_string();
+
+        subdomain
+    }
+}
+
+/// Represents a cloud provider and a region.
+/// Useful to transform a user input selecting a
+/// cloud provider region to an enum and vice-versa.
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq)]
+pub enum CloudProviderRegion {
+    /// Represents `aws/us-east-1` cloud provider and region
+    #[serde(rename = "aws/us-east-1")]
+    AwsUsEast1,
+    /// Represents `aws/eu-west-1` cloud provider and region
+    #[serde(rename = "aws/eu-west-1")]
+    AwsEuWest1,
+}
+
+impl CloudProviderRegion {
+    /// Converts a CloudProvider object to a CloudProviderRegion.
+    pub fn from_cloud_provider(cloud_provider: CloudProvider) -> Result<Self, Error> {
+        match cloud_provider.id.as_str() {
+            "aws/us-east-1" => Ok(CloudProviderRegion::AwsUsEast1),
+            "aws/eu-west-1" => Ok(CloudProviderRegion::AwsEuWest1),
+            _ => Err(Error::CloudProviderRegionParseError),
+        }
+    }
+}
+
+impl Display for CloudProviderRegion {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            CloudProviderRegion::AwsUsEast1 => write!(f, "aws/us-east-1"),
+            CloudProviderRegion::AwsEuWest1 => write!(f, "aws/eu-west-1"),
+        }
+    }
+}
+
+impl FromStr for CloudProviderRegion {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "aws/us-east-1" => Ok(CloudProviderRegion::AwsUsEast1),
+            "aws/eu-west-1" => Ok(CloudProviderRegion::AwsEuWest1),
+            _ => Err(Error::CloudProviderRegionParseError),
+        }
+    }
+}
+
+impl Client {
+    /// List all the available cloud providers.
+    ///
+    /// E.g.: [us-east-1, eu-west-1]
+    pub async fn list_cloud_providers(&self) -> Result<Vec<CloudProvider>, Error> {
+        #[derive(Deserialize)]
+        #[serde(rename_all = "camelCase")]
+        struct CloudProviderResponse {
+            data: Vec<CloudProvider>,
+            #[serde(skip_serializing_if = "Option::is_none")]
+            next_cursor: Option<String>,
+        }
+        let mut cloud_providers: Vec<CloudProvider> = vec![];
+        let mut cursor: String = String::new();
+
+        loop {
+            let req = self
+                .build_request(Method::GET, ["api", "cloud-regions"], "sync")
```

Comment:
```suggestion
                .build_request(Method::GET, ["api", "cloud-regions"], format!("https://sync.{}", self.endpoint))
```

(May need to do some more finagling to get the "https://" in the right spot.)